lib/web_ui/lib/src/engine/js_interop/js_promise.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@JS()
+library js_promise;
+
+import 'package:js/js.dart';
+
+/// Type-safe JS Promises
+@JS('Promise')
```

Comment:
What happens if any other code writes this same JS-interop @JS('Promise')?

Reply:
Well, there's at least this: https://github.com/flutter/engine/blob/0cb8d6d9858c510fc488f7fc738d68922a52f857/lib/web_ui/lib/src/engine/safe_browser_api.dart#L254